python/prairielearn.py

```diff
@@ -15,7 +15,42 @@
 import importlib.util
 import os
 import collections
-
+from typing import Dict, Any, TypedDict, Literal, Optional
+from typing_extensions import NotRequired
+
+class PartialScore(TypedDict):
+    "A class with type signatures for the partial scores dict"
+    score: Optional[float]
+    weight: NotRequired[int]
+    feedback: NotRequired[str]
```

Comment:
`Optional` means the key must exist in the dictionary, but its value can be `None`, but `NotRequired` means that the key doesn't even have to exist in the dictionary, is that right? Is there documentation that this is matches what is expected for this dictionary? I'm not sure myself.

Reply:
That's correct. I actually don't know what the partial score dictionary is required to have or not have, I just copied this over from the type signature in our repo. @nwalters512 is there any documentation for what's required to be in this dictionary?